org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -164,11 +164,12 @@ public <T> CompletableFuture<Optional<T>> computeFirst(BiFunction<? super Langua
 		return result.thenApplyAsync(Function.identity());
 	}
 
-	public @NonNull E withComparator(final @Nullable Comparator<? super LanguageServerWrapper> comparator) {
-		this.comparator = comparator;
+	public @NonNull E withPreferredServer(final @Nullable String serverId) {
```

Comment:
Could we use a LanguageServerDefiniton here? Stronger typing usually makes code less error-prone.

Reply:
the consumer does not have the `LanguageServerDefiniton`, it only has the id, so I think that for this PR is it fine to leave it as this.

Reply:
The consumer can get the languageServerDefinition from the id (`LanguageServerRegistry.getDefinition(id)`). Really, one never regrets stronger typing.

Reply:
Ok, done